From d392444806c576bbae57f31f6e8457f0b560ed24 Mon Sep 17 00:00:00 2001
From: Shawn Xiao <b49994@freescale.com>
Date: Wed, 28 Jan 2015 16:39:16 +0800
Subject: [PATCH 1476/1543] MLK-10169 GPU: fix kernel booting issue on iMX6sx
 SOC without GPU

commit 7fe4c7e7f9fe7c92fef1c3f42da62bfc571e3f57 from
git://git.freescale.com/imx/linux-2.6-imx.git

Add checking GPU module logic in qos init. This prevents kernel
booting issue in the iMX6sx SOC where there is no GPU module.

Signed-off-by: Shawn Xiao <b49994@freescale.com>
---
 arch/arm/mach-imx/mach-imx6sx.c |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/arch/arm/mach-imx/mach-imx6sx.c b/arch/arm/mach-imx/mach-imx6sx.c
index 3b3da52..7f5188f 100755
--- a/arch/arm/mach-imx/mach-imx6sx.c
+++ b/arch/arm/mach-imx/mach-imx6sx.c
@@ -239,6 +239,10 @@ static inline void imx6sx_qos_init(void)
 	struct device_node *np;
 	void   __iomem *src_base;
 
+	np = of_find_compatible_node(NULL, NULL, "fsl,imx6sx-gpu");
+	if (!np || !of_device_is_available(np))
+		return;
+
 	np = of_find_compatible_node(NULL, NULL, "fsl,imx6sx-qosc");
 	if (!np)
 		return;
-- 
1.7.5.4

